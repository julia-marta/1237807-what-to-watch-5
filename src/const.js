export const ALL_GENRES = `All genres`;
export const FILM_CARDS_COUNT_TO_SHOW = 8;

export const TABS = [`Overview`, `Details`, `Reviews`];

export const Tab = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

export const AppRoute = {
  ROOT: `/`,
  LOGIN: `/login`,
  MY_LIST: `/mylist`,
  FILMS: `/films`,
  REVIEW: `/review`,
  PLAYER: `/player`,
};

export const APIRoute = {
  FILMS: `/films`,
  PROMO: `/films/promo`,
  FAVORITE: `/favorite`,
  COMMENTS: `/comments`,
};
