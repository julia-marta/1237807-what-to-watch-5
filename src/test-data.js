import React from 'react';
import PropTypes from "prop-types";
export const noop = () => {};

export const promoFilm = {
  id: 1,
  name: `The Grand Budapest Hotel`,
  posterImage: `the-grand-budapest-hotel-poster.jpg`,
  backgroundImage: `bg-the-grand-budapest-hotel.jpg`,
  backgroundColor: `#000000`,
  genre: `Drama`,
  released: 2014,
  isFavorite: false,
};

export const filmCard = {
  id: 1,
  name: `The Grand Budapest Hotel`,
  genre: `Drama`,
  previewImage: `the-grand-budapest-hotel.jpg`,
  previewVideoLink: `https://trailers.apple.com/movies/universal/krampus/krampus-tlr1_h480p.mov`,
};

export const film = {
  id: 1,
  name: `The Grand Budapest Hotel`,
  posterImage: `the-grand-budapest-hotel-poster.jpg`,
  backgroundImage: `bg-the-grand-budapest-hotel.jpg`,
  backgroundColor: `#000000`,
  genre: `Drama`,
  released: 2014,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  rating: 8.9,
  scoresCount: 240,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  runTime: 99,
  isFavorite: true,
};

export const reviews = [
  {
    id: 1,
    user: {
      id: 4,
      name: `Kate Muir`,
    },
    rating: 8.9,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`,
  },
  {
    id: 2,
    user: {
      id: 5,
      name: `Matthew Lickona`,
    },
    rating: 7.2,
    comment: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    date: `2019-05-18T14:13:56.569Z`,
  }
];

export const genres = [`All genres`, `Drama`, `Music`, `Biography`, `Survival`, `Thriller`, `Crime`];

export const defaultState = {
  DATA: {
    films: [filmCard, filmCard, filmCard, filmCard, filmCard, filmCard, filmCard],
    promo: promoFilm,
    currentFilm: film,
    currentReviews: reviews,
  },
  FILTER: {
    genre: `Drama`,
    cardsCount: 8,
  },
  USER: {
    status: `NOT_AUTHORIZED`,
    name: `userName`,
    avatar: `avatar.jpg`,
    reviewStatus: `NOT_SAVING`,
  },
};

export const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

export const SimpleMockComponent = () => <div />;
