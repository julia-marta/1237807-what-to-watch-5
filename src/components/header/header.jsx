import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {Link} from "react-router-dom";
import Logo from "../logo/logo";
import {redirectToRoute} from "../../store/actions/user-actions/user-actions";
import {AppRoute, AuthorizationStatus} from "../../const";

const {ROOT, LOGIN, MY_LIST} = AppRoute;
const {AUTHORIZED} = AuthorizationStatus;

const Header = (props) => {

  const {isMain, isSignIn, classTitle, children, userStatus, userName, userAvatar, onAvatarClick} = props;

  return (
    <header className={`page-header ${classTitle ? `${classTitle}` : ``}`}>
      <div className="logo">
        {isMain ? <a className="logo__link">
          <Logo/>
        </a>
          : <Link to={ROOT} className="logo__link">
            <Logo/>
          </Link>}
      </div>

      {children}

      {isSignIn ? `` :
        <div className="user-block">

          {userStatus === AUTHORIZED ?
            <div className="user-block__avatar" onClick={() => onAvatarClick(MY_LIST)}>
              <img src={userAvatar} alt={userName} width="63" height="63" />
            </div>

            : <Link to={LOGIN} className="user-block__link">Sign in</Link>}
        </div>
      }

    </header>
  );
};

Header.propTypes = {
  isMain: PropTypes.bool,
  isSignIn: PropTypes.bool,
  classTitle: PropTypes.string,
  children: PropTypes.element,
  userStatus: PropTypes.string.isRequired,
  userName: PropTypes.oneOfType([PropTypes.string.isRequired, () => null]),
  userAvatar: PropTypes.oneOfType([PropTypes.string.isRequired, () => null]),
  onAvatarClick: PropTypes.func.isRequired,
};

const mapStateToProps = ({USER}) => ({
  userStatus: USER.status,
  userName: USER.name,
  userAvatar: USER.avatar,
});

const mapDispatchToProps = (dispatch) => ({
  onAvatarClick(url) {
    dispatch(redirectToRoute(url));
  }
});

export {Header};
export default connect(mapStateToProps, mapDispatchToProps)(Header);
