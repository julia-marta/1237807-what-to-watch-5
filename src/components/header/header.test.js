import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from 'react-router-dom';
import {Header} from "./header";
import {noop, defaultState} from "../../test-data";
import {AuthorizationStatus} from "../../const";

const {name, avatar} = defaultState.USER;
const {AUTHORIZED, NOT_AUTHORIZED} = AuthorizationStatus;

describe(`should Header render correctly`, () => {
  it(`on Main Page`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <Header isMain={true} onAvatarClick={noop} userStatus={AUTHORIZED} userName={name} userAvatar={avatar} />
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`on SignIn Page`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <Header isSignIn={true} onAvatarClick={noop} userStatus={NOT_AUTHORIZED} userName={null} userAvatar={null}/>
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`when user is authorized`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <Header onAvatarClick={noop} userStatus={AUTHORIZED} userName={name} userAvatar={avatar} />
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`when user is not authorized`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <Header onAvatarClick={noop} userStatus={NOT_AUTHORIZED} userName={null} userAvatar={null} />
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
