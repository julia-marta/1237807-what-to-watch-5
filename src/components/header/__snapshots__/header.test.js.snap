// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should Header render correctly on Main Page 1`] = `
<header
  className="page-header "
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <div
      className="user-block__avatar"
      onClick={[Function]}
    >
      <img
        alt="userName"
        height="63"
        src="avatar.jpg"
        width="63"
      />
    </div>
  </div>
</header>
`;

exports[`should Header render correctly on SignIn Page 1`] = `
<header
  className="page-header "
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  
</header>
`;

exports[`should Header render correctly when user is authorized 1`] = `
<header
  className="page-header "
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <div
      className="user-block__avatar"
      onClick={[Function]}
    >
      <img
        alt="userName"
        height="63"
        src="avatar.jpg"
        width="63"
      />
    </div>
  </div>
</header>
`;

exports[`should Header render correctly when user is not authorized 1`] = `
<header
  className="page-header "
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <a
      className="user-block__link"
      href="/login"
      onClick={[Function]}
    >
      Sign in
    </a>
  </div>
</header>
`;
